test/Test/Database/LSMTree/Normal/StateMachine.hs

```diff
@@ -1776,7 +1777,11 @@ runActionsBracket' ::
   -> Actions (Lockstep state) -> QC.Property
 runActionsBracket' p init cleanup runner tagger actions =
     tagFinalState actions tagger
-  $ Lockstep.Run.runActionsBracket p init cleanup runner actions
+  $ Lockstep.Run.runActionsBracket p init cleanup' runner actions
+  where
+    cleanup' st = do
+      cleanup st
+      checkForgottenRefs
```

Comment:
Arguably, `checkForgottenRefs` only needs to be included when we are running the real implementation, but I think it's also fine to keep it here

Reply:
Yes only necessary. But may still actually work in io-sim (but I've not yet checked).